core/src/main/scala/org/apache/spark/storage/BlockManagerMasterEndpoint.scala

```diff
@@ -510,14 +514,20 @@ class BlockManagerMasterEndpoint(
   }
 
   /**
-   * Returns the BlockManagerId with topology information populated, if available.
+   * Returns an option of BlockManagerId. If topology information is available, it populated inside
+   * BlockManagerId. If the BlockManager belongs to a recently removed executor, None is returned.
    */
   private def register(
       idWithoutTopologyInfo: BlockManagerId,
       localDirs: Array[String],
       maxOnHeapMemSize: Long,
       maxOffHeapMemSize: Long,
-      storageEndpoint: RpcEndpointRef): BlockManagerId = {
+      storageEndpoint: RpcEndpointRef): Option[BlockManagerId] = {
+    if (Option(recentlyRemovedExecutors.getIfPresent(idWithoutTopologyInfo.executorId)).isDefined) {
```

Comment:
Nit: for just checking the object is null you do not need to build an Option (Option is good when when you pass the object to another method to emphasize it can be null).